nurse_scheduling.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2020 D-Wave Systems Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This code includes an implementation of the algorithm described in Ikeda, 
+# K., # Nakamura, Y. & Humble, T.S. Application of Quantum Annealing to Nurse 
+# Scheduling Problem. Sci Rep 9, 12837 (2019). 
+# https://doi.org/10.1038/s41598-019-49172-3, © The Author(s) 2019, use of 
+# which is licensed under a Creative Commons Attribution 4.0 International 
+# License (To view a copy of this license, visit 
+# http://creativecommons.org/licenses/by/4.0/).
+
+from dwave.system import LeapHybridSampler
+from dimod import BinaryQuadraticModel
+
+# count nurses n = 1 ... N_NURSES
+# count scheduling days as d = 1 ... N_DAYS
+# binary variable q_nd is the assignment of nurse n to day d
+# a is a positive correlation coefficient for implementing the hard nurse
+# constraint - value provided by Ikeda, Nakamura, Humble
+a = 3.5
+N_NURSES = 3
+N_DAYS = 11
+SIZE = N_DAYS * N_NURSES
+
+# Hard shift constraint: at least one nurse is working each day
+# implemented by penalizing any nurses scheduled on two successive
+# days
+
+# Hard nurse constraint: no nurse works two or more consecutive days
+# Lagrange parameter, for hard nurse constraint, on workforce and effort
+# Workforce function W(d) - set to a constant for now
+# Effort function E(n) - set to a constant for now
+lagrange_parameter = 1.3
+W = 1
+E = 1
+
+# Soft nurse constraint: all nurses should have approximately even work
+#                        schedules
+# Lagrange parameter, for shift constraints, on work days
+# preference function G - set to a constant for now
+# Minimum duty days F - the number of work days that each nurse wants
+# to be scheduled. At present, each will do the minimum on average.
+# The parameter gamma's value suggested by Ikeda, Nakamura, Humble
+gamma = 0.3
+G = 1
+F = int(N_DAYS/N_NURSES)
+
+
+def indx(n, d):
+    return n * N_DAYS + d
+
+
+def get_nurse_and_day(indx):
+    return (int(indx / N_DAYS), indx % N_DAYS)
+
+
+# Hard shift constraint - does not have Lagrange parameter - J matrix
```

Comment:
`Hard shift constraint` on its own, doesn't really mean much to the audience. Instead, perhaps add that helpful comment from your README? (i.e. `Hard shift constraint - at least one nurse for each working day`)

Reply:
Thanks. I did this.